diff -ru AnnotSV-2.5.original/Makefile AnnotSV-2.5.patched/Makefile
--- AnnotSV-2.5.original/Makefile	2020-10-13 09:02:59.000000000 +0000
+++ AnnotSV-2.5.patched/Makefile	2020-10-13 16:05:45.024128586 +0000
@@ -40,9 +40,9 @@
 VERSION              := 2.5
 RM                   := /bin/rm
 RMDIR                := /bin/rmdir
-MKDIR                := install -d
+MKDIR                := install -d -m 2775
 MV                   := /bin/mv
-CP                   := install -p -m 0644
+CP                   := install -p -m 0664
 CPDIR                := /bin/cp -r
 CONFIGFILE           := etc/$(ANNOTSV)/configfile
 MAKEFILE             := Makefile
@@ -80,13 +80,13 @@
 	@echo "Configfile configuration"
 	@echo "------------------------"
 	$(MKDIR) $(DESTDIR)$(ETCDIR)/$(ANNOTSV)
-	install -p -m 0755 $(CONFIGFILE)  $(DESTDIR)$(ETCDIR)/$(ANNOTSV)
+	install -p -m 0664 $(CONFIGFILE)  $(DESTDIR)$(ETCDIR)/$(ANNOTSV)
 
 install-makefile: $(MAKEFILE)
 	@echo ""
 	@echo "Makefile installation"
 	@echo "---------------------"
-	install -p -m 0755 $(MAKEFILE)  $(DESTDIR)$(PREFIX)
+	install -p -m 0664 $(MAKEFILE)  $(DESTDIR)$(PREFIX)
 
 install-executable:
 	@echo ""
@@ -152,7 +152,7 @@
 	$(RM) -rf 2003_hg19.tar.gz
 
 install-exomiser-2:
-	install -p -m 0755 $(PROPERTIES) $(DESTDIR)$(ETCDIR)/$(ANNOTSV)
+	install -p -m 0664 $(PROPERTIES) $(DESTDIR)$(ETCDIR)/$(ANNOTSV)
 	$(CPDIR) share/AnnotSV/jar/ $(DESTDIR)$(SHAREDIR)/$(ANNOTSV)/
 
 install-exomiser-3:
