diff -ru demuxlet-3ab507cf7c0489f983fe69b2c6c3ccdd6d853729.original/Makefile.am demuxlet-3ab507cf7c0489f983fe69b2c6c3ccdd6d853729.patched/Makefile.am
--- demuxlet-3ab507cf7c0489f983fe69b2c6c3ccdd6d853729.original/Makefile.am	2019-12-12 19:54:15.000000000 +0000
+++ demuxlet-3ab507cf7c0489f983fe69b2c6c3ccdd6d853729.patched/Makefile.am	2020-04-17 14:24:59.797236509 +0000
@@ -3,7 +3,7 @@
 ## to provide a way for the user to supply additional arguments.
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-AM_CPPFLAGS = -I ../htslib/ -I ../../htslib/htslib -pipe -D__STDC_LIMIT_MACROS -Wall -Wno-unused-local-typedefs -Wno-enum-compare -fpic -O2 
+AM_CPPFLAGS = -I /apps/software/HTSlib/1.9-GCCcore-7.3.0/include/htslib -pipe -D__STDC_LIMIT_MACROS -Wall -Wno-unused-local-typedefs -Wno-enum-compare -fpic -O2 
 
 noinst_HEADERS = \
 	Constant.h pException.h \
@@ -24,4 +24,4 @@
 	tsv_reader.cpp \
 	cmd_cram_demuxlet.cpp
 
-demuxlet_LDADD = ../htslib/libhts.a -lpthread -llzma -lz -lbz2 -lgomp -lcurl -lcrypto
+demuxlet_LDADD = /apps/software/HTSlib/1.9-GCCcore-7.3.0/lib/libhts.a -lpthread -llzma -lz -lbz2 -lgomp -lcurl -lcrypto
