diff -ru EPACTS-3.3.2.original/src/chaps.cpp EPACTS-3.3.2.patched/src/chaps.cpp
--- EPACTS-3.3.2.original/src/chaps.cpp	2017-02-22 08:30:03.000000000 +0000
+++ EPACTS-3.3.2.patched/src/chaps.cpp	2020-08-12 18:31:21.100771531 +0000
@@ -40,8 +40,8 @@
   bool sepchr;
 
   static int const DEFAULT_NSNPS = 10000L;
-  static double const DEFAULT_MIN_MAF = 1e-6;
-  static double const DEFAULT_MAX_MAF = 1;
+  static double constexpr DEFAULT_MIN_MAF = 1e-6;
+  static double constexpr DEFAULT_MAX_MAF = 1;
 
   pCHAPSArgs() :
     unit(DEFAULT_NSNPS), verbose(false), ignoreFilter(false), ignoreMissing(false), minAC(0), minMAC(0), maxAC(INT_MAX), minMAF(DEFAULT_MIN_MAF), maxMAF(DEFAULT_MAX_MAF), minCallRate(DEFAULT_MIN_MAF), sepchr(false)
diff -ru EPACTS-3.3.2.original/src/FreqTable.h EPACTS-3.3.2.patched/src/FreqTable.h
--- EPACTS-3.3.2.original/src/FreqTable.h	2017-02-22 08:30:03.000000000 +0000
+++ EPACTS-3.3.2.patched/src/FreqTable.h	2020-08-12 18:31:39.852064818 +0000
@@ -15,7 +15,7 @@
     };
     void remove(const T& t) {
         if (this->data.find(t) == this->data.end()) {
-            return false;
+            return;
         }
         this->data[t] -- ;
         this->isSorted = false;
diff -ru EPACTS-3.3.2.original/src/pEmmax.h EPACTS-3.3.2.patched/src/pEmmax.h
--- EPACTS-3.3.2.original/src/pEmmax.h	2017-02-22 08:30:03.000000000 +0000
+++ EPACTS-3.3.2.patched/src/pEmmax.h	2020-08-12 18:30:31.284195416 +0000
@@ -83,8 +83,8 @@
   double maxP;
 
   static int const DEFAULT_UNIT = 10000L;
-  static double const DEFAULT_MIN = 1e-6;
-  static double const DEFAULT_MAX = 1;
+  static double constexpr DEFAULT_MIN = 1e-6;
+  static double constexpr DEFAULT_MAX = 1;
   static int const DEFAULT_MIN_MAC = 1;
   static int const DEFAULT_MAX_MAC = 1000000000;
   static int const DEFAULT_MAX_PERM = 10000000;
@@ -114,8 +114,8 @@
   static const int N_MAGIC = 8;
   static const char *magicKin;
   static const char *magicEig;
-  static const double ZEPS = 1e-10;
-  static const double TOL = 1e-6;
+  static constexpr double ZEPS = 1e-10;
+  static constexpr double TOL = 1e-6;
 
   static bool writeIDs(wFile& wf, std::vector<std::string>& ids) {
     char* p = (char*) malloc(65536); // maximum ID length is 64k
diff -ru EPACTS-3.3.2.original/src/vcfast.cpp EPACTS-3.3.2.patched/src/vcfast.cpp
--- EPACTS-3.3.2.original/src/vcfast.cpp	2017-02-22 08:30:03.000000000 +0000
+++ EPACTS-3.3.2.patched/src/vcfast.cpp	2020-08-12 18:30:58.672999896 +0000
@@ -64,8 +64,8 @@
   bool sepchr;
 
   static int const DEFAULT_UNIT = 10000L;
-  static double const DEFAULT_MIN = 1e-6;
-  static double const DEFAULT_MAX_MAF = 1;
+  static double constexpr DEFAULT_MIN = 1e-6;
+  static double constexpr DEFAULT_MAX_MAF = 1;
 
   pVPHArgs() :
     unit(DEFAULT_UNIT), verbose(false), ignoreFilter(false), ignoreMissing(false), includeMultiAllelic(false), minAC(0), minMAC(0), maxAC(INT_MAX), minMAF(DEFAULT_MIN), maxMAF(DEFAULT_MAX_MAF), minCallRate(DEFAULT_MIN), genoFlag(false), acFlag(false), anFlag(false), aldFlag(false), tstvFlag(false),sepchr(false)
